src/Microsoft.Dnx.Compilation.CSharp.Common/MetadataReferenceExtensions.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Reflection.PortableExecutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.Dnx.Compilation.Caching;
+
+namespace Microsoft.Dnx.Compilation.CSharp
+{
+    /// <summary>
+    /// Extension methods for <see cref="IMetadataReference"/>
+    /// </summary>
+    public static class MetadataReferenceExtensions
+    {
+        /// <summary>
+        /// Converts an <see cref="IMetadataReference"/> to a <see cref="MetadataReference"/> instance.
+        /// </summary>
+        /// <param name="metadataReference">The <see cref="IMetadataReference"/> to convert.</param>
+        /// <param name="cache">The <see cref="ICache"/>.</param>
+        /// <param name="onAssemblyMetadataCacheMiss">Invoked when a cache miss occurs when attempting to look 
+        /// up cached <see cref="AssemblyMetadata"/>.</param>
+        /// <returns>The converted <see cref="MetadataReference"/>.</returns>
+        public static MetadataReference ConvertMetadataReference(
+            this IMetadataReference metadataReference,
+            ICache cache,
+            Action<CacheContext, string> onAssemblyMetadataCacheMiss)
```

Comment:
It's not the prettiest, but given that this is purely an internal helper that we expect few people to use, I'm not that afraid of it. It's cleaner than bringing FileWriteTimeCacheDependency along when we don't need it.